frontend/src/components/features/settings/api-keys-manager.tsx

```diff
@@ -0,0 +1,318 @@
+    <>
+      <div className="flex flex-col gap-6">
+        <div className="flex items-center justify-between">
+          <BrandButton
+            type="button"
+            variant="primary"
+            onClick={() => setCreateModalOpen(true)}
+          >
+            {t(I18nKey.SETTINGS$CREATE_API_KEY)}
+          </BrandButton>
+        </div>
+
+        <p className="text-sm text-gray-300">
+          {t(I18nKey.SETTINGS$API_KEYS_DESCRIPTION)}
+        </p>
+
+        {isLoading && (
+          <div className="flex justify-center p-4">
+            <LoadingSpinner size="large" />
+          </div>
+        )}
+        {!isLoading && Array.isArray(apiKeys) && apiKeys.length > 0 && (
+          <div className="border border-tertiary rounded-md overflow-hidden">
+            <table className="w-full">
+              <thead className="bg-base-tertiary">
+                <tr>
+                  <th className="text-left p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$NAME)}
+                  </th>
+                  <th className="text-left p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$KEY_PREFIX)}
+                  </th>
+                  <th className="text-left p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$CREATED_AT)}
+                  </th>
+                  <th className="text-left p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$LAST_USED)}
+                  </th>
+                  <th className="text-right p-3 text-sm font-medium">
+                    {t(I18nKey.SETTINGS$ACTIONS)}
+                  </th>
+                </tr>
+              </thead>
+              <tbody>
+                {apiKeys.map((key) => (
+                  <tr key={key.id} className="border-t border-tertiary">
+                    <td className="p-3 text-sm">{key.name}</td>
+                    <td className="p-3 text-sm font-mono">{key.prefix}...</td>
+                    <td className="p-3 text-sm">
+                      {formatDate(key.created_at)}
+                    </td>
+                    <td className="p-3 text-sm">
+                      {formatDate(key.last_used_at)}
+                    </td>
+                    <td className="p-3 text-right">
+                      <button
+                        type="button"
+                        className="underline"
+                        onClick={() => {
+                          setKeyToDelete(key);
+                          setDeleteModalOpen(true);
+                        }}
+                      >
+                        {t(I18nKey.BUTTON$DELETE)}
+                      </button>
+                    </td>
+                  </tr>
+                ))}
+              </tbody>
+            </table>
+          </div>
+        )}
+      </div>
+
+      {/* Create API Key Modal */}
+      {createModalOpen && (
+        <ModalBackdrop>
+          <div
+            data-testid="create-api-key-modal"
```

Comment:
NIT: move these logic chunks for create, delete, and show new key could be helpful

Reply:
@openhands please handle this feedback

Reply:
Hmm something went wrong with this job

Reply:
@openhands please split the chunks of logic in this file for create, delete, and show new key into separate components